packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts

```diff
@@ -143,11 +153,27 @@ export class SagemakerTrainTask implements ec2.IConnectable, sfn.IStepFunctionsT
         });
 
         // add the security groups to the connections object
-        if (this.props.vpcConfig) {
-            this.props.vpcConfig.securityGroups.forEach(sg => this.connections.addSecurityGroup(sg));
+        if (props.vpcConfig) {
+            this.vpc = props.vpcConfig.vpc;
+            this.securityGroup = new ec2.SecurityGroup(scope, 'TrainJobSecurityGroup', {
+                vpc: this.vpc
+            });
+            this.connections = new ec2.Connections({ securityGroups: [this.securityGroup] });
+            this.securityGroups.push(this.securityGroup);
+            this.subnets = (props.vpcConfig.subnets) ? (props.vpcConfig.subnets.map(s => (s.subnetId))) : this.vpc.selectSubnets().subnetIds;
         }
     }
 
+    /**
+     * Add the security group to all instances via the launch configuration
+     * security groups array.
+     *
+     * @param securityGroup: The security group to add
+     */
+    public addSecurityGroup(securityGroup: ec2.ISecurityGroup): void {
+        this.securityGroups.push(securityGroup);
```

Comment:
Just so you know, this will not add it to the `Connections` object automatically, so it will not have SG rules applied. Is that intended?

Reply:
There is always a SecurityGroup created when VPC is supplied as a param. This will be updated where there is a connection with other resources. This method is to allow extra Security Groups to be added to the config